### PR TITLE
 [slang-tidy] Add quiet options and IDs to slang-tidy checks

### PR DESCRIPTION
**Pull request by Sustrak**
This MR adds improvements in the slang-tidy by adding new CLI options to control verbosity, adds an explicit ID to the checks.

Also fixes a bug which made the feature of avoiding checks for certain files to not work properly

### CODE REVIEW

File: tools/tidy/src/tidy.cpp [hunk 1]
```diff
@@ -144,28 +200,51 @@ int main(int argc, char** argv) {
     // Check all enabled checks
     for (const auto& checkName : Registry::getEnabledChecks()) {
         const auto check = Registry::create(checkName);
-        OS::print(fmt::format("[{}]", check->name()));
 
-        driver.diagEngine.setMessage(check->diagCode(), check->diagString());
+        if (!quiet)
```
Comment (MikePopoloski) [inline tools/tidy/src/tidy.cpp:204]: Note that for all of these std::optional<bool> arguments you're checking whether they've been provided, not whether they are set to true. Someone could still do something like `--quiet=false` and the tool would instead treat it like quiet was set to true. You actually want `.value_or(false)` or `quiet != true` or something like that.

Comment (Sustrak) [reply tools/tidy/src/tidy.cpp:202]: Didn't know about the `--option=value` trick. Fixed :D

File: tools/tidy/src/tidy.cpp [hunk 2]
```diff
@@ -124,13 +178,15 @@ int main(int argc, char** argv) {
     }
     SLANG_CATCH(const std::exception& e) {
 #if __cpp_exceptions
-        slang::OS::printE(fmt::format("internal compiler error: {}\n", e.what()));
+        if (!superQuiet)
```
[the remark this answers is not in the document]
Comment (Sustrak) [reply tools/tidy/src/tidy.cpp:180]: Agree

[the remark this answers is not in the document]
Comment (MikePopoloski) [reply tools/tidy/src/tidy.cpp:180]: Hmm, you marked as resolved but the check is still here.

[the remark this answers is not in the document]
Comment (Sustrak) [reply tools/tidy/src/tidy.cpp]: Sorry, forgot to add the changes before commiting

**Pull request merged**